whisper/whisperv6/peer_test.go

```diff
@@ -332,34 +388,52 @@ func TestPeerBasic(t *testing.T) {
 	}
 }
 
-func powReqExchange(t *testing.T) {
+func checkPowExchangeForNodeZero(t *testing.T) {
+	cnt := 0
 	for i, node := range nodes {
 		for peer := range node.shh.peers {
-			if peer.powRequirement > 1000.0 {
-				t.Fatalf("node %d: one of the peers' pow requirement is too big (%f).", i, peer.powRequirement)
+			if peer.peer.ID() == discover.PubkeyID(&nodes[0].id.PublicKey) {
+				cnt++
+				if peer.powRequirement != masterPow {
+					t.Fatalf("node %d: failed to set the new pow requirement.", i)
+				}
 			}
 		}
 	}
+	if cnt == 0 {
+		t.Fatalf("no matching peers found.")
+	}
+}
 
-	const pow float64 = 7777777.0
-	nodes[0].shh.SetMinimumPoW(pow)
-
-	// wait until all the messages are delivered
-	time.Sleep(64 * time.Millisecond)
-
-	cnt := 0
+func checkPowExchange(t *testing.T) {
 	for i, node := range nodes {
 		for peer := range node.shh.peers {
-			if peer.peer.ID() == discover.PubkeyID(&nodes[0].id.PublicKey) {
-				cnt++
-				if peer.powRequirement != pow {
-					t.Fatalf("node %d: failed to set the new pow requirement.", i)
+			if peer.peer.ID() != discover.PubkeyID(&nodes[0].id.PublicKey) {
+				if peer.powRequirement != masterPow {
+					t.Fatalf("node %d: failed to exchange pow requirement in round %d; expected %f, got %f",
+						i, round, masterPow, peer.powRequirement)
 				}
 			}
 		}
 	}
+}
+
+func checkBloomFilterExchange(t *testing.T) {
+	for i, node := range nodes {
+		for peer := range node.shh.peers {
+			if !isBloomFilterEqual(peer.bloomFilter, masterBloomFilter) {
+				t.Fatalf("node %d: failed to exchange bloom filter requirement in round %d. \n%x expected \n%x got",
+					i, round, masterBloomFilter, peer.bloomFilter)
+			}
+		}
+	}
+}
 
-	if cnt == 0 {
-		t.Fatalf("no matching peers found.")
+func isBloomFilterEqual(a, b []byte) bool {
```

Comment:
This function should be available in the general api, not just in a test file.

Reply:
indeed